docs/guidelines/git-signed-commits.md

```diff
@@ -0,0 +1,172 @@
+
+!!! Note Signing method
+
+    We recommend using self-signed SSH keys for signing your git commits (In the future we may switch to a certificate based approach)
+
+## Configure your local development environment
+
+For this guideline our reference is using Git from the command line. 
+
+We assume that Git and SSH is installed on your system. 
+
+We assume that you are using SSH to authenticate Git with github.com. Consult our [guideline](git-github.md) for more information on this topic.
+
+The Github documentation on [SSH commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification) is a good source for detailed information.
+
+### Adding a SSH key for signing
+
+!!! Note Use multiple SSH keys
+
+    We recommend using different SSH keys for authentication and signing. This may add some extra complexity and it may provide a more robust set-up with looser coupling of key components of the SDLC.
+
+To create a new SSH key for signing you can use the following command (alter the date manually):
+
+```shell
+ssh-keygen -t ed25519 -f ~/.ssh/git_ssh_signing_key_1 -C "Created on <date>, for larskaare on github.com"
+```
+
+This will create a SSH signing key and add a comment on date and purpose. Add a passphrase to the key. Success full key generation will output the key fingerprint and a randomart image (randomart is supposed to be a visualisation making it easier to validate keys - and identify changes)
+
+Add the new key to the ssh-agent
+
+```shell
+ssh-add ~/.ssh/git_ssh_signing_key_1
+```
+
+### [Configure Git](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-ssh-key) to use the SSH key
+
+We will configure the git global settings to use the new SSH key for signing commits locally. Examples assume you created the key as defined above.
+
+```shell
+git config --global gpg.format ssh
+git config --global user.signingkey $HOME/.ssh/git_ssh_signing_key_1.pub
+git config --global commit.gpgsign true
+```
+
+These lines will tell git to use SSH for signing commit, tell git where to find the key that should be used and then tell git to always sign commits. If you do not add this last line you will specifically have to add the `-S` parameter for each commit you can to sign.
+
+### Examining the git log
+
+To verify that commits are signed locally you can use the following command:
+
+```shell
+git log --show-signature
+```
+
+When you run this command on a newly configured system you may get an error message like `error: gpg.ssh.allowedSignersFile needs to be configured and exist for ssh signature verification` In order for git to verify signatures locally you need to add the public keys that are used to sign to a file that Git will use.
```

Comment:
The error message might be a bit "elusive" to see, as git will output the log too and the error itself is not easily seen in the wall of text.
Should we comment on that, as the error might be easily overlooked...